functional-tests/mining_chain_test.go

```diff
@@ -87,7 +86,8 @@ func TestSyncFromSingleMiner(t *testing.T) {
 	presealPath := filepath.Join(wd, "..", "fixtures/genesis-sectors")
 	genTime := int64(1000000000)
 	blockTime := 30 * time.Second
-	fakeClock := th.NewFakeClock(time.Unix(genTime, 0))
+	// The clock is set some way ahead of the genesis block.
+	fakeClock := clock.NewFake(time.Unix(genTime, 0).Add(4 * time.Hour))
```

Comment:
I don't think this we should add this edge case as the default to our syncing tests, there should be one test that covers this behavior.

Reply:
I've put this back to the "right" time, with advances when mining. Note that the clock is pretty much irrelevant to block production here since the MineOnce call avoids the scheduler and just sets the "right" block time. But the validating nodes chain sync will still reject blocks in the future.

It would be better if we could avoid MineOnce, or get it down to an entry into the scheduler rather than avoiding it.